File: TcpSocketSystem_DotNet/src/UseCases/GestionarSesiones/IGestionarSesionesUseCase.cs
using TcpSocketSystem.Core.Domain.Modelos;

namespace TcpSocketSystem.Core.UseCases.GestionarSesiones;

/// <summary>
/// Caso de Uso: Gestionar el Ciclo de Vida y Concurrencia de Sesiones TCP.
/// </summary>
public interface IGestionarSesionesUseCase
{
    EstadoSesion RegistrarNuevaConexion(string sesionId, string direccionIp, int puerto);
    void RegistrarActividadMensaje(string sesionId);
    void RegistrarDesconexion(string sesionId, string motivo = "Desconexión normal");
    int ObtenerConexionesActivas();
}

File: TcpSocketSystem_Java/src/com/tecnm/sockets/usecases/gestionarsesiones/IGestionarSesionesUseCase.java
package com.tecnm.sockets.usecases.gestionarsesiones;

import com.tecnm.sockets.domain.modelos.EstadoSesion;

/**
 * Caso de Uso: Control de ciclo de vida de sesiones y concurrencia.
 */
public interface IGestionarSesionesUseCase {

    EstadoSesion registrarNuevaConexion(String sesionId, String direccionIp, int puerto);

    void registrarActividadMensaje(String sesionId);

    void registrarDesconexion(String sesionId, String motivo);

    int obtenerConexionesActivas();
}
